### PR TITLE
feat(unlock-app) - improve AddressInput validation

### PR DESCRIPTION
**Pull request by kalidiagne**
# Description

- [x] Improve `AddressInput` validation 

# Issues

Fixes #
Refs #

# Checklist:

- [ ] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [ ] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

File: packages/ui/lib/components/Form/AddressInput.stories.tsx [hunk 1]
```diff
@@ -40,6 +42,13 @@ const Template: ComponentStory<typeof AddressInput> = () => {
     console.log('error value', error)
   }
 
+  const isAddressOrEns = (address = '') => {
+    return (
+      address?.toLowerCase()?.includes('.eth') ||
```
Comment (julien51) [inline packages/ui/lib/components/Form/AddressInput.stories.tsx:47]: ENS are actually not limited to `.eth`.

File: unlock-app/src/utils/strings.tsx [hunk 2]
```diff
@@ -37,3 +39,11 @@ export const getValidNumber = (value: string | number): number | undefined => {
     ? parseInt(`${value}`)
     : undefined
 }
+
+/** Check if string is an address or ens */
+
+export const isAddressOrEns = (address = '') => {
```
Comment (julien51) [inline unlock-app/src/utils/strings.tsx:45]: Let's just export the other one so we don't duplicate logic?

Comment (kalidiagne) [reply unlock-app/src/utils/strings.tsx:45]: sure, you mean from the component or the other in codebase?

Comment (julien51) [reply unlock-app/src/utils/strings.tsx:45]: From the component. And if we use in either places just make sure we only have a single way of doing this.

File: packages/ui/lib/components/Form/AddressInput.stories.tsx [hunk 3]
```diff
@@ -31,6 +32,7 @@ const Template: ComponentStory<typeof AddressInput> = () => {
   })
 
   const onSubmit = (form: any, e: any) => {
+    console.log('form')
```
Comment (julien51) [inline packages/ui/lib/components/Form/AddressInput.stories.tsx:35]: Remove please

File: packages/ui/lib/utils.ts [hunk 4]
```diff
@@ -6,3 +6,9 @@ export const minifyAddress = (address: string) => {
     ? `${address.slice(0, 5)}...${address.slice(address.length - 5)}`
     : address
 }
+
+export const isAddressOrEns = (address = '') => {
+  return (
+    address?.toLowerCase()?.includes('.eth') || ethers.utils.isAddress(address)
```
Comment (julien51) [inline packages/ui/lib/utils.ts:12]: Kali, ENS names don't necessarily end up with `.eth`.... So I don't think that's right. If you want to check if something is a valid ENS you need to check that it resolves correctly.

Comment (kalidiagne) [reply packages/ui/lib/utils.ts:12]: making this an async function will cause a side-effect in the component, the idea here is to have a result without calling `resolveName` and waiting for the result.

**Comment by julien51** [comment]
https://github.com/unlock-protocol/unlock/issues/11092

**Comment by julien51** [comment]
For now let's just use `.eth` so we can ship and we can add more resolution later

**Review by julien51 — APPROVED** [review]

**Pull request merged**